pkg/gen/node-templates/provider.ts.mustache

```diff
@@ -18,19 +18,23 @@ export namespace {{Group}} {
 
       {{/Properties}}
 
+      public getInputs(): inputApi.{{Group}}.{{Version}}.{{Kind}} { return this.__inputs; }
+      private readonly __inputs: inputApi.{{Group}}.{{Version}}.{{Kind}};
+
       /**
       * Create a {{Group}}.{{Version}}.{{Kind}} resource with the given unique name, arguments, and options.
       *
       * @param name The _unique_ name of the resource.
       * @param args The arguments to use to populate this resource's properties.
       * @param opts A bag of options that control this resource's behavior.
       */
-      constructor(name: string, args?: inputApi.{{Group}}.{{Version}}.{{Kind}}, opts?: pulumi.ResourceOptions) {
+      constructor(name: string, args: inputApi.{{Group}}.{{Version}}.{{Kind}}, opts?: pulumi.ResourceOptions) {
           let inputs: pulumi.Inputs = {};
           {{#Properties}}
           inputs["{{Name}}"] = {{{DefaultValue}}};
```

Comment:
Is this necessary still, now that `args` is required? I see in the generated code this will generate `args ? args.prop : undefined` even though `args` can't be undefined now.

Also, are we OK with the breaking change here? I'm fine with it, as long as we're aware of it.

Reply:
Resolved offline. We can get rid of the ternary operator, but `{{{DefaultValue}}}` needs to stay because the Mustache generation code needs to munge `inputs["apiVersion"]`.